lib/Link.sc

```diff
@@ -0,0 +1,38 @@
+/*
+ * Link: A stage in a signal flow chain.
+ *
+ * A Link is an abstract representation of an element or phase in a
+ * more complicated signal flow processing chain. A Link is initialized
+ * with certain settings passed into the constructor, then the transform
+ * function is called. The transform function accepts an audio signal
+ * and outputs another audio signal. This allows for transformation to be
+ * chained together.
+ *
+ * var signal = SoundIn(inputChannel);
+ * signal = Link1().transform(signal);
+ * signal = Link2().transform(signal);
+ * signal = Link3().transform(signal);
+ * Out(outputChannel, signal)
+ * 
+ * This can be thought of as:
+ *
+ * Input Signal => Link 1 => Link 2 => Link 3 => Output
+ * 
+ * The transformations that links can do to signals may vary. A Link
+ * may make assumptions about the input signals, including whether or not
+ * they are arrays comprised of individual audio channels. Links may also
+ * modify the signals by changing their dimensionality. For example, the
+ * AggregateLink performs a sum operation to effectively flatten a 2D array.
+ * Additionally, transformations may invoke multichannel expansion, depending
+ * on the parameters and input signals.
+ */
+Link : Class {
```

Comment:
I think `Link` should be thought of as a more configurable item, though I do agree that it's a little unclear even to me where the exact distinction lies. If a `Link` is not a `UGen` then we should probably not use `.ar`. If a `Link` is a `UGen` then `.ar` makes sense to use. 

At least in my mind, I think of a `Link` as something that has a somewhat higher level of abstraction than a `UGen`, is more configurable, and can add some extra complexity. What we are calling `functions` might better be implemented as something like `SignalChain` classes which are objects comprised of `Link` objects strung together. In this sense, the difference seems more semantic than functional, but I do think it helps in terms of making it easier to read and understand the code. But it could also be feasible to make all of our `Link` objects into custom `UGen` objects.